android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -169,6 +186,32 @@ constructor(
       ?.also { configsLoadedCallback(true) }
   }
 
+  fun fetchNonWorkflowConfigResources() {
+    CoroutineScope(dispatcherProvider.io()).launch {
+      try {
+        Timber.i("Fetching non-workflow resources for app $appId")
+        repository
+          .searchCompositionByIdentifier(appId)
+          ?.section
+          ?.groupBy { it.focus.reference.split("/")[0] }
+          ?.entries
+          ?.filterNot {
+            it.key == ResourceType.Binary.name || it.key == ResourceType.Parameters.name
+          }
+          ?.forEach { entry: Map.Entry<String, List<Composition.SectionComponent>> ->
+            val ids = entry.value.joinToString(",") { it.focus.extractId() }
+            val rPath = entry.key + "?${Composition.SP_RES_ID}=$ids"
```

Comment:
Please use more descriptive variable names. This helps make the code readable now that we do not document all methods.

Reply:
This has been updated